src/components/common/Content/MovieTVShowPageContent/MovieTVShowPageContent.tsx

```diff
@@ -0,0 +1,61 @@
+import './MovieTVShowPageContent.css'
+import { useLocation } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { MOVIE_PAGE_CONTENT_KEY } from "../../../../data/local-storage-keys.ts";
+import { MovieTvPageContent } from "../../../../interfaces /MoviePageResponseInterface.ts";
+
+export const MovieTVShowPageContent = () => {
+    const location = useLocation()
+    const { id, type } = location.state
+    const [movieTvPageData, setMovieTvPageData] = useState((): MovieTvPageContent => {
+        const storedData = localStorage.getItem(MOVIE_PAGE_CONTENT_KEY);
+        return storedData ? JSON.parse(storedData)[0] : null;
+    })
+
+    useEffect(() => {
+        const itemId = id ? parseInt(id) : null
+
+        if (movieTvPageData && itemId === movieTvPageData.id) {
+            const storedData = localStorage.getItem(MOVIE_PAGE_CONTENT_KEY);
+
+            if (storedData && JSON.parse(storedData).length === 0) {
+                setMovieTvPageData(JSON.parse(storedData)[0]);
+            }
```

Comment:
JSOM parsing twice. JSON parsing might be long time operation so better do it once.